metron-interface/metron-rest/src/main/java/org/apache/metron/rest/controller/RestExceptionHandler.java

```diff
@@ -35,7 +36,7 @@
   @ResponseBody
   ResponseEntity<?> handleControllerException(HttpServletRequest request, Throwable ex) {
     HttpStatus status = getStatus(request);
-    return new ResponseEntity<>(new RestError(status.value(), ex.getMessage(), getFullMessage(ex)), status);
+    return new ResponseEntity<>(new RestError(status.value(), ex.getMessage(), ExceptionUtils.getStackTrace(ex)), status);
```

Comment:
Why change to stack trace from message?
ExceptionUtils.getRootCauseMessage() ?

Reply:
Because @cestella requested it earlier.

Reply:
ok, but I thought you wanted to show the root cause to the user, **and** have the stack trace that casey wanted.  Not just the stack trace.  What is the user going to see now?

Reply:
I would prefer just root cause but I had the impression the full stack trace was wanted.  Doesn't make that much difference to me so we just need to agree on what it should be.

Reply:
Well, the stack trace is generally more helpful in debugging than just the root cause, so yeah, I kinda wanted the stack trace along with the root cause.  I'm willing to be argued with, but as a dev, I'd like to know line numbers of failures along with why.

Reply:
Are you ok with having the full stack trace returned @ottobackwards?

Reply:
My preference would be -> root cause only.  Stack trace if profile is not production.  Stack trace to log.

I will not hold this back for that, but if this gets audited, that is what is going to have to be done later.

Reply:
I mean root cause ALWAYS, stack trace if profile allows

Reply:
My vote would be root cause in the REST response, full stack trace in the logs.  I would also vote to skip the separate profile for stack trace.  Turning that on and restarting REST would be more effort than just looking at the log.  Doubt anyone would use it.

I think this is a minor issue.  I would be good with either.

Reply:
I agree with you @merrimanr

Reply:
I think I can live with just the root cause, but I'd like to know how exposing the stack trace is a security issue first.  Can you clarify the reasoning behind it, @ottobackwards ?  It's not that I disbelieve you, but I'd like to better understand because we currently have stack traces in logs all over the place.

Reply:
https://software-security.sans.org/blog/2010/08/11/security-misconfigurations-java-webxml-files

Reply:
https://www.owasp.org/index.php/Error_Handling#Vulnerable_Patterns_for_Error_Handling

Reply:
http://www.creativebloq.com/web-design/website-security-tips-protect-your-site-7122853

etc etc.

Reply:
It is not in logs that is the problem, it is if it is in the UI.  I think that it would apply to rest too... but I am not sure on that

Reply:
That's fine; I'm with it :)  Thanks for the clarity, @ottobackwards